backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/pageindicator/BpkPageIndicator.kt

```diff
@@ -0,0 +1,143 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.pageindicator
+
+import androidx.compose.animation.animateColorAsState
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.itemsIndexed
+import androidx.compose.foundation.lazy.rememberLazyListState
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.platform.LocalDensity
+import androidx.compose.ui.semantics.invisibleToUser
+import androidx.compose.ui.semantics.semantics
+import androidx.compose.ui.unit.Dp
+import kotlinx.coroutines.launch
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+import java.lang.Integer.min
+
+enum class BpkPageIndicatorStyle {
+  Default,
+  OverImage,
+}
+
+@OptIn(ExperimentalComposeUiApi::class)
+@Composable
+fun BpkPageIndicator(
+  currentIndex: Int,
+  totalIndicators: Int,
+  modifier: Modifier = Modifier,
+  style: BpkPageIndicatorStyle = BpkPageIndicatorStyle.Default,
+) {
+
+  if (totalIndicators <= 1) throw IllegalArgumentException("totalIndicators must be greater than 1")
+  if (currentIndex !in 0 until totalIndicators) throw IllegalArgumentException("currentIndex must be between 0 and $totalIndicators")
+  val indicatorSize = BpkSpacing.Md
+  val indicatorBoxWidth = indicatorSize * 2
+  val indicatorBoxHeight = indicatorSize * 3
+  val indicatorCount = min(totalIndicators, DISPLAY_DOTS_MAX)
+  val rowWidth = (indicatorBoxWidth * indicatorCount) + indicatorSize
+
+  val coroutineScope = rememberCoroutineScope()
+  val state = rememberLazyListState()
+
+  val offsetCount = when {
```

Comment:
What does this variable mean? The amount of dots before the current one? nit: Can we simplify the naming and the way we calculate it?

Reply:
I probably understand what got me confused here – this variable is a multiplier for dot width and not item width, therefore instead of 1, 3,4 you use 2, 6, 8 etc. Let's simplify it to work with items count